### PR TITLE
New logging system

### PR DESCRIPTION
**Pull request by papadop**
This adds a new logging system that is better (in my view) than passing a verbose bool on all relevant function. Basically, we have a log_stream(XX) function that provides a logging stream which is decided on the value of XXX to be either std::cout or a null stream. Several logging levels are defined but only two are exploited at this time (INFORMATION and PROGRESS).

The class Logger keeps (with a static member the global state of the logger. For now, I have disabled the logging globally for python (see __init__.py), but it may be meaningful to disable it only for mne....

Compared to the previous solution, it also includes the progressbars.
So far, I kept this only for OpenMEEG (i.e not OpenMEEGMaths).

### CODE REVIEW

File: wrapping/python/openmeeg/__init__.py [hunk 1]
```diff
@@ -2,3 +2,6 @@
 from .openmeeg import *  # noqa
 from ._version import __version__
 from ._make_geometry import make_geometry, make_nested_geometry
+from openmeeg import Logger, WARNING
```
[the remark this answers is not in the document]
Comment (larsoner) [reply wrapping/python/openmeeg/__init__.py:5]:
```suggestion
```

**Comment by larsoner** [comment]
> For now, I have disabled the logging globally for python (see init.py), but it may be meaningful to disable it only for mne....

Eventually we should just add a Python function like `set_log_level` that calls `Logger.logger().set_log_level` under the hood, but we can add that later I think. I'll mark this as merge-when-green, then hopefully it'll be easy to merge into #582 !

**Comment by papadop** [comment]
Thank's. I was hesitating at using the #noqa. Thought, I was missing some pythonic way of doing things....
Hopefully, everything will pass now.

**Comment by papadop** [comment]
Good idea, but handling the list might be complicated....
But if it's manageable, I agree...

**Pull request merged**